doc/entropy-seems-wrong.md

```diff
@@ -0,0 +1,107 @@
+
+
+### Character Set Sizes
+
+This math is possible because we know how many letters are in our alphabet.  To include uppercase letters and lowercase letters the set grows to `26 + 26 = 52` options for each letter in the password.  If we were dealing with an unknown language or if we didn't want to group the letters into sets of characters then we have a harder time.
+
+There are a couple options that the password strength software has taken to combat this problem.  The first is using a technique from Claude Shannon, where the letters in the password itself are used as the character set size.  Using the password of "aaaabbbb" would indicate there's only 2 options per character and we'd have `2^8` possibilities.  This works way better for longer passwords that have more varied characters.
+
+The other technique is to lump letters together into groups.  When a letter is encountered from a group, the entire group's size is included.  A lot of systems require you to use "one capital letter, one number, one symbol" and our character sets are similar.  So the password of "aA0!aA0!" would score poorly with Shannon's technique (`4^8` possibilities) but much better when using the groups of letters (`26 + 26 + 10 + 10 = 72` options for each character, `72^8` possibilities).
+
+
+### Character Prediction
+
+When checking passwords, one should be encouraged (though not required) to use randomly generated passwords that do not look like words.  If you typically use words, those passwords should be weaker.  To accomplish this, a large amount of text was analyzed for letter trigraphs, which is essentially saying "when given two letters, what's the chance you can predict the third letter?"
+
+Let's take a look at the sample password "thimbles".  If we analyze the first two letters, we can predict what the next letter will be.  There's about a 67% chance that "e" will follow "th".  There's about a 10% chance it could be an "a" and 9% chance a word ends with "th".  Our letter is "i", and there's only about a 6.5% chance that "thi" will be the outcome.  So, 93.5% of the time we will be wrong.
+
+Our entropy calculations would say that the maximum score for the third letter would be 26, but we only keep 93.5% of that value, resulting in a lower overall entropy score.
+
+
+Hacking Techniques
+------------------
+
+Really, the password strength meter does not measure the technique used to generate a password.  It only measures the password and what it would take for the password to be compromised.  Explained another way: your password generator may have a truly random technique of making your password but that doesn't help if it generates "abcd1234".
+
+
+### Common Passwords
+
+People collect lists of passwords that have been successful on other sites.  This will be the first attack that will be performed whenever someone is trying to hack or crack passwords.
```

Comment:
Don't need that int he second sentence.

"This will be the first attack performed whenever someone is trying to hack or crack passwords."